src/registry/registryAccess.ts

```diff
@@ -71,9 +74,31 @@ export class RegistryAccess {
    * @returns An array of metadata type objects that require strict parent folder names
    */
   public getStrictFolderTypes(): MetadataType[] {
-    return Object.values(this.registry.strictDirectoryNames).map(
-      (typeId) => this.registry.types[typeId]
-    );
+    if (!this.strictFolderTypes) {
+      this.strictFolderTypes = Object.values(this.registry.strictDirectoryNames).map(
+        (typeId) => this.registry.types[typeId]
+      );
```

Comment:
We only need to do this once.  The registry data can't change once RegistryAccess is created.